modules/library/referencing/src/main/java/org/geotools/referencing/operation/matrix/Matrix3.java

```diff
@@ -31,7 +31,6 @@
  * @source $URL$
  * @version $Id$
  * @author Martin Desruisseaux (IRD)
- * @deprecated Use GeneralMatrix
  */
```

Comment:
> About Matrix3 and Matrix4, it seems that vecmath had special high performance classes for the 3 and 4 cases, while
here we have replaced them with a general construct instead, so we are going to pay array checks for all operations.
Not awfully worried, the cases in which we do a ton of computations are for coverage reprojection, where we tend to replace the full math with a localized affine equivalent that's coded in JAI, but ... I thought you were going to expect a speedup, by what I see, maybe we'll get a small slowdown instead?

Interestingly I found the "fast" implementation we could be using behind this class - FixedMatrix2x2_64F. Looks like there is a host of auto generated implementations for fixed matrixes that we could be using instead of DenseMatrix64F.